### DISCUSSION TITLE
Code showing correctly in preview, but not actual page?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#23726] [Other Feature Feedback, Questions, & Ideas]
<p><a href="https://github.com/fintechntrusts2020/Replica-Economics" rel="noopener nofollow ugc">https://github.com/fintechntrusts2020/Replica-Economics</a> , I have another post where I’m still having issues with inserting images, however, now my bullet points are not correct as they were. Help?</p>

**Comment by otter-computer** [comment]
<p>Can you post a direct link to the page you’re having trouble with and maybe share a screenshot of what the issue is/what it’s meant to look like?</p>

**Comment by ghost** [comment]
<p>Preview page link; <a href="https://github.com/fintechntrusts2020/Replica-Economics/blob/532f492f3fecd0bba1f3b92388968a7a28fb60df/_pages/origin.md" rel="noopener nofollow ugc">https://github.com/fintechntrusts2020/Replica-Economics/blob/532f492f3fecd0bba1f3b92388968a7a28fb60df/_pages/origin.md</a></p>
<p>However, below is the link to the live page and it shows a broken image</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      
      <a href="https://fintechntrusts2020.github.io/Replica-Economics/origin" target="_blank" rel="noopener nofollow ugc">fintechntrusts2020.github.io</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:647/500;"><img src="https://user-images.githubusercontent.com/70776696/181078923-86dcb601-f9d9-4217-bdcd-d7534c2ad93a.jpeg" class="thumbnail" width="647" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/4/94e971fefb017f71922f14eb5cfbffbdb6081148_2_647x500.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/9/4/94e971fefb017f71922f14eb5cfbffbdb6081148_2_970x750.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/9/4/94e971fefb017f71922f14eb5cfbffbdb6081148_2_1294x1000.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/4/94e971fefb017f71922f14eb5cfbffbdb6081148_2_10x10.png"></div>

<h3><a href="https://fintechntrusts2020.github.io/Replica-Economics/origin" target="_blank" rel="noopener nofollow ugc">Why Replica Exists? – Replica Economics</a></h3>

<p>The history behind Replica</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by ghost** [comment]
<p>This is the current command for the image &lt;“img src=”/images/demo/meeting.jpg"&gt;</p>

**Comment by otter-computer** [comment]
<p>Thanks for posting those links! The reason this is happening is because of a relative path issue with how the browser loads this image file on <a href="http://GitHub.com">GitHub.com</a> compared to on your Pages site. Looking at the code for this image:</p>
<pre><code class="lang-auto">&lt;img src="/images/demo/meeting.jpg"&gt;
</code></pre>
<p>The source for this image starts with a <code>/</code> character. This tells the browser to search for the image file starting at the root of the domain. When you’re browsing <a href="http://GitHub.com">GitHub.com</a> this means it will look in the root of the repository then follow the file path to the correct file.</p>
<p>However, on your Pages site, there’s an extra element that’s required in your URL—the repository name <code>Replica-Economics</code>.</p>
<p>The actual URL that your image is loaded from is this:</p>
<pre><code class="lang-nohighlight">https://fintechntrusts2020.github.io/Replica-Economics/images/demo/meeting.jpg
</code></pre>
<p>However, currently the browser is looking for it at this URL:</p>
<pre><code class="lang-nohighlight">https://fintechntrusts2020.github.io/images/demo/meeting.jpg
</code></pre>
<p>This is happening because the initial <code>/</code> character is telling the browser to look for this path at the root of the domain (<code>https://fintechntrusts2020.github.io/</code>), instead of looking for this path relative to the current open page (<code>https://fintechntrusts2020.github.io/Replica-Economics/</code>), meaning the browser will leave out the required <code>Replica-Economics</code> “directory” in the URL.</p>
<p>To fix this you’ll simply need to remove the <code>/</code> character from the beginning of the image’s source path:</p>
<pre><code class="lang-auto">&lt;img src="images/demo/meeting.jpg"&gt;
</code></pre>
<p>If you’re having other issues with images like this it’s likely that there’s other relative pathing issues causing similar problems!</p>

**Comment by ghost** [comment]
<p>Hey there thomas, yes your advice worked. However, the link is now broken on the preview, is that purely because of a relative pathing issue? If so, where my be a logical area to begin troubleshooting? Index, config, etc?</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Oop, I missed this part during my testing, sorry! :see_no_evil:</p>
<p>Looks like there is a way to fix it for both at the same time. You’ll first need to convert this <code>img</code> tag to a markdown image tag, and add back in the initial <code>/</code>:</p>
<pre><code class="lang-markdown">![](/images/demo/meeting.jpg)
</code></pre>
<p>Then you’ll need to make sure you set your <code>baseurl</code> to the name of your repository inside your <code>_config.yml</code> file:</p>
<pre><code class="lang-yaml">baseurl: /Replica-Economics
</code></pre>
<p>Once you do both of these your image should show up correctly on both GitHub and on your website!</p>
